app/app.tsx

```diff
@@ -46,6 +54,38 @@ interface AppProps {
 // Creating a react-query client with the stale time set to "Infinity" so that its never stale
 const queryClient = new QueryClient({ defaultOptions: { queries: { staleTime: Infinity } } })
 
+// Setting up our custom Toast component
+const CustomToast = () => {
+  const insets = useSafeAreaInsets()
+
+  useLayoutEffect(() => {
+    // handle a new push notification received while the app is in "foreground" state
+    const unsubscribe = messaging().onMessage(async (remoteMessage) => {
+      if (remoteMessage.notification) {
+        Toast.show({
+          text1: remoteMessage.notification.title,
```

Comment:
Does the Firebase dashboard require title and body? Or at least one? 

If not, maybe here we should check if both are empty so we don't send out blank push notifications

Reply:
good point. updated the logic here: [955300f](https://github.com/infinitered/ChainReactApp2023/pull/36/commits/955300f3b195ed60d5449a26734e9af839b69236)

`Toast` does a good job of handling if one or the other is `undefined`.